src/plugins/accounting.cpp

```diff
@@ -404,3 +404,253 @@ json_t* convert_projects_to_json (const std::vector<std::string> projects)
 
     return known_projects;
 }
+
+int load_associations (
+                    json_t *data,
+                    std::map<int, std::map<std::string, Association>> &users,
+                    std::map<int, std::string> &users_def_bank)
+{
+    char *bank, *def_bank, *assoc_queues, *assoc_projects, *def_project = NULL;
+    int uid, max_running_jobs, max_active_jobs, max_nodes, max_cores = 0;
+    double fshare = 0.0;
+    json_error_t error;
+    int num_data = 0;
+    int active = 1;
+
+    if (!data || !json_is_array (data)) {
+        return -1;
+    }
+    num_data = json_array_size (data);
+
+    for (int i = 0; i < num_data; i++) {
+        json_t *el = json_array_get(data, i);
+
+        if (json_unpack_ex (el, &error, 0,
+                            "{s:i, s:s, s:s, s:F, s:i, s:i, s:s, s:i, s:s, "
+                            "s:s, s:i, s:i}",
+                            "userid", &uid,
+                            "bank", &bank,
+                            "def_bank", &def_bank,
+                            "fairshare", &fshare,
+                            "max_running_jobs", &max_running_jobs,
+                            "max_active_jobs", &max_active_jobs,
+                            "queues", &assoc_queues,
+                            "active", &active,
+                            "projects", &assoc_projects,
+                            "def_project", &def_project,
+                            "max_nodes", &max_nodes,
+                            "max_cores", &max_cores) < 0) {
+            return -1;
+        }
```

Comment:
The `json_error_t` is unused here - should any jansson error be returned to the caller or logged here? (Same goes for the other unpacks)

Reply:
Yeah, this is a good call. Thanks for pointing this out; I think maybe I should try to just log an error with `flux_log_error`, perhaps to `stderr` since this error would be occurring during plugin load - what do you think?

Reply:
I don't think logging to `stderr` is appropriate. Logging an error to the flux log seems right.